crates/common/consensus/src/misc.rs

```diff
@@ -10,3 +10,131 @@ pub fn compute_signing_root<SSZObject: TreeHash>(ssz_object: SSZObject, domain:
     }
     .tree_hash_root()
 }
+use std::cmp::max;
+
+use alloy_primitives::aliases::B32;
+use anyhow::ensure;
+use ethereum_hashing::hash;
+
+use crate::{
+    deneb::beacon_state::BeaconState,
+    fork_choice::helpers::constants::{
+        MAX_EFFECTIVE_BALANCE, MAX_SEED_LOOKAHEAD, SHUFFLE_ROUND_COUNT, SLOTS_PER_EPOCH,
+    },
+    fork_data::ForkData,
+};
+
+static MAX_RANDOM_BYTE: u64 = 255;
+
+pub fn compute_shuffled_index(
+    index: usize,
+    index_count: usize,
+    seed: B256,
+) -> anyhow::Result<usize> {
+    ensure!(index < index_count, "Index must be less than index_count");
+    let mut index = index;
+    for round in 0..SHUFFLE_ROUND_COUNT {
+        let seed_with_round = [seed.as_slice(), &round.to_le_bytes()].concat();
+        let pivot = bytes_to_int64(&hash(&seed_with_round)[..]) % index_count as u64;
+
+        let flip = (pivot as usize + (index_count - index)) % index_count;
+        let position = max(index, flip);
+        let seed_with_position =
+            [seed_with_round.as_slice(), &(position / 256).to_le_bytes()].concat();
+        let source = hash(&seed_with_position);
+        let byte = source[(position % 256) / 8];
+        let bit = (byte >> (position % 8)) % 2;
+
+        index = if bit == 1 { flip } else { index };
+    }
+    Ok(index)
+}
+
+fn bytes_to_int64(slice: &[u8]) -> u64 {
+    let mut bytes = [0; 8];
+    bytes.copy_from_slice(&slice[0..8]);
+    u64::from_le_bytes(bytes)
+}
+
+/// Return from ``indices`` a random index sampled by effective balance
+pub fn compute_proposer_index(
+    state: BeaconState,
+    indices: &[u64],
+    seed: B256,
+) -> anyhow::Result<u64> {
+    ensure!(!indices.is_empty(), "Index must be less than index_count");
+
+    let mut i: usize = 0;
+    let total = indices.len();
+
+    loop {
+        let candidate_index = indices[compute_shuffled_index(i % total, total, seed)?];
+
+        let seed_with_index = [seed.as_slice(), &(i / 32).to_be_bytes()].concat();
```

Comment:
```suggestion
        let seed_with_index = [seed.as_slice(), &(i / 32).to_le_bytes()].concat();
```

Is this supposed to be big endian? I thought it is supposed to be little endian